lib/bridge/IrcBridge.js

```diff
@@ -754,6 +769,128 @@ IrcBridge.prototype.getBotClient = function(server) {
     });
 }
 
+
+var _solveTimeout = null;
+
+IrcBridge.prototype._updateVisibilityMap = function(isMode, key, value) {
+    let hasChanged = false;
+    if (isMode) {
+        if (typeof value !== 'boolean') {
+            throw new Error('+s state must be indicated with a boolean');
+        }
+        if (this._visibilityMap.channelModes[key] !== value) {
+            this._visibilityMap.channelModes[key] = value;
+            hasChanged = true;
+        }
+    }
+    else {
+        if (typeof value !== 'string' || (value !== "private" && value !== "public")) {
+            throw new Error('Room visibility must = "private" | "public"');
+        }
+
+        if (this._visibilityMap.roomVisibilities[key] !== value) {
+            this._visibilityMap.roomVisibilities[key] = value;
+            hasChanged = true;
+        }
+    }
+
+    if (hasChanged) {
+        clearTimeout(_solveTimeout);
+
+        _solveTimeout = setTimeout(() => {
+            this._solveVisibility()
+        }, 10000);
```

Comment:
I'm also not sure how much we want to *always* assign-then-solve here. When the bridge starts up, it could be getting thousands of these guys, it may be better to separate the calls here (so `_updateVisibilityMap` returns `hasChanged`), and let the caller decide if they want to solve (which they may only do after startup, after they have processed a block of users, whatever).